internal/cmd/controller/target/status_test.go

```diff
@@ -0,0 +1,504 @@
+package target
+
+import (
+	"testing"
+
+	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// availableTarget returns a Target that is considered available.
+func availableTarget() *Target {
+	return &Target{
+		Deployment: &fleet.BundleDeployment{
+			Spec: fleet.BundleDeploymentSpec{
+				DeploymentID:       "id",
+				StagedDeploymentID: "id",
+			},
+			Status: fleet.BundleDeploymentStatus{
+				AppliedDeploymentID: "id",
+				Ready:               true,
+			},
+		},
+		DeploymentID: "id",
+	}
+}
+
+// unavailableTargetMismatchedID returns a Target that is considered
+// unavailable due to a mismatched ID.
+func unavailableTargetMismatchedID() *Target {
+	return &Target{
+		Deployment: &fleet.BundleDeployment{
+			Spec: fleet.BundleDeploymentSpec{
+				DeploymentID:       "id",
+				StagedDeploymentID: "id",
+			},
+			Status: fleet.BundleDeploymentStatus{
+				AppliedDeploymentID: "off-id",
+				Ready:               true,
+			},
+		},
+		DeploymentID: "id",
+	}
+}
+
+// unavailableTargetNonReady returns a Target that is considered
+// unavailable due to not being ready.
+func unavailableTargetNonReady() *Target {
+	return &Target{
+		Deployment: &fleet.BundleDeployment{
+			Spec: fleet.BundleDeploymentSpec{
+				DeploymentID:       "id",
+				StagedDeploymentID: "id",
+			},
+			Status: fleet.BundleDeploymentStatus{
+				AppliedDeploymentID: "id",
+				Ready:               false,
+			},
+		},
+		DeploymentID: "id",
+	}
+}
+
+func targetWithRolloutStrategy(target *Target, rolloutStrategy fleet.RolloutStrategy) *Target {
+	if target.Bundle == nil {
+		target.Bundle = &fleet.Bundle{}
+	}
+	target.Bundle.Spec.RolloutStrategy = &rolloutStrategy
+	return target
+}
+
+func Test_limit(t *testing.T) {
+	tests := []struct {
+		name    string
+		count   int
+		val     []*intstr.IntOrString
+		want    int
+		wantErr bool
+	}{
+		{
+			name:  "fixed value below count",
+			count: 10,
+			val: []*intstr.IntOrString{
+				{IntVal: 5},
+			},
+			want: 5,
+		},
+		{
+			name:  "fixed value above count",
+			count: 10,
+			val: []*intstr.IntOrString{
+				{IntVal: 15},
+			},
+			want: 15,
```

Comment:
In effect, wouldn't this be equivalent in behaviour to having the result be 10?
As in, specifying a maximum of 15 unavailable clusters for a partition containing 10 of them will never result in more than 10 unavailable clusters for that partition.

Just a thought on the implementation, not on the test case itself.